src/components/pages/Calendar/RenderCalendar.js

```diff
@@ -1,90 +1,164 @@
-import React from 'react';
+import React, { useState } from 'react';
 import DashHeader from '../../common/DashHeader';
 import SearchIcon from './SearchIcon';
-import { Layout, DatePicker, Form, Button, Input, Space } from 'antd';
+import {
+  Layout,
+  DatePicker,
+  Form,
+  Button,
+  Input,
+  Space,
+  Calendar,
+  ConfigProvider,
+  List,
+  Modal,
+} from 'antd';
 import moment from 'moment';
+import 'antd/dist/antd.dark.css';
 
 const { Sider, Content, Footer } = Layout;
 const weekFormat = 'MM/DD';
 const customWeekStartEndFormat = value =>
-  `${moment(value)
-    .startOf('week')
-    .format(weekFormat)} ~ ${moment(value)
+  `${moment(value).startOf('week').format(weekFormat)} ~ ${moment(value)
     .endOf('week')
     .format(weekFormat)}`;
 
 function RenderCalendar() {
-  const createEvent = event => {
-    console.log('The button is working');
+  const [selectedDate, setSelectedDate] = useState(moment());
+  const [events, setEvents] = useState([]); // Initialize an empty array to store events
+  const [isModalVisible, setIsModalVisible] = useState(false); // Initialize the modal visibility state
+  const [newEvent, setNewEvent] = useState({
+    title: '',
+    date: moment(),
+  }); // Initialize the new event data state
+
+  const handleDateChange = value => {
+    setSelectedDate(value);
+  };
+
+  const handleCreateEventClick = () => {
+    setIsModalVisible(true); // Show the modal when the "CREATE EVENT" button is clicked
+  };
+
+  const handleModalOk = () => {
+    const eventData = {
+      title: newEvent.title,
+      date: newEvent.date,
+      // TODO: Add other event data fields as needed
+    };
+    setEvents([...events, eventData]); // Add the new event data to the events array
+    setIsModalVisible(false); // Hide the modal when the user submits the event details
+    setNewEvent({
+      // Reset the new event data state
+      title: '',
+      date: moment(),
+    });
+    console.log('New event created:', eventData); // Log the new event data to the console
   };
 
-  const searchEvent = event => {
-    console.log('The button is working');
+  const handleModalCancel = () => {
+    setIsModalVisible(false);
+    setNewEvent({
+      title: '',
+      date: moment(),
+    });
   };
 
-  const calendars = event => {
-    console.log('The button is working');
```

Comment:
I'm one of those who likes commented code, but this line is useless now, don't you think ?

Reply:
Yes, it is but until we get the validation and BE tied we need a test point.